MerkleTreeConstructor.java

```diff
@@ -0,0 +1,99 @@
+package com.karolismed.simple_blockchain.blockchain;
+
+import com.karolismed.simple_blockchain.blockchain.model.transaction.Transaction;
+import com.karolismed.simple_blockchain.hashing.HashingService;
+import lombok.NonNull;
+import org.bitcoinj.core.Sha256Hash;
+import org.bitcoinj.core.Utils;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.stream.Collectors;
+
+public class MerkleTreeConstructor {
+
+    private HashingService hashingService;
+
+    public MerkleTreeConstructor() {
+        this.hashingService = new HashingService();
+    }
+
+    public String getMerkleTreeRoot(@NonNull List<Transaction> transactions) {
+        return getMerkleTreeRootFromIds(
+            transactions.stream().map(Transaction::getTxId).collect(Collectors.toList())
+        );
+    }
+
+    public String getMerkleTreeRootFromIds(@NonNull List<String> transactionHashes) {
+        List<Sha256Hash> reversedHashes = transactionHashes.stream()
+            .map(hash -> Sha256Hash.wrapReversed(Utils.HEX.decode(hash.toLowerCase())))
```

Comment:
It would be advisable to use Locale when performing toLowerCase call.

So the ending of the statement would be hash.toLowerCase(Locale.ENGLISH)

Reply:
Addressed